### PR TITLE
feat(core-text-area): upgrade to styled-components

### PR DESCRIPTION
**Pull request by marcod1419**
Converts `TextArea` to styled-components and improves its docs.

### CODE REVIEW

File: packages/TextArea/TextArea.jsx [hunk 1]
```diff
@@ -1,40 +1,212 @@
...
+      </Box>
+    )
+  }
 )
 
+TextArea.displayName = 'TextArea'
+
 TextArea.propTypes = {
+  /**
+   * The id. If no `id` is provided, a default `id` will be generated in the format of: "`name`\_`value`".
```
Comment (theetrain) [inline packages/TextArea/TextArea.jsx]: Please add a clearer meaning of the generated ID's output, using an example.

File: packages/TextArea/TextArea.jsx [hunk 2]
```diff
@@ -54,7 +226,7 @@ TextArea.propTypes = {
   /**
    * An error message. Either an error or a helper should be used, not both.
    */
-  error: PropTypes.string,
+  error: PropTypes.node,
```
Comment (theetrain) [inline packages/TextArea/TextArea.jsx]: Why was this changed to node?

File: packages/TextArea/package.json [hunk 3]
```diff
@@ -20,20 +20,25 @@
...
     "prop-types": "^15.5.10"
   },
   "devDependencies": {
-    "@tds/core-colours": "^1.1.2",
-    "@tds/core-input-feedback": "^1.0.13",
-    "@tds/core-paragraph": "^1.0.12",
-    "@tds/core-standalone-icon": "^1.1.0",
-    "@tds/core-text": "^1.1.0",
-    "@tds/shared-animation": "^1.2.0"
+    "react-transition-group": "^2.2.1"
```
Comment (theetrain) [inline packages/TextArea/package.json]: Is this being used anywhere?

File: packages/TextArea/TextArea.jsx [hunk 4]
```diff
@@ -1,40 +1,212 @@
...
+  colorShuttleGrey,
+  colorWhite,
+  colorPrimary,
+  colorCardinal,
+  colorAthensGrey,
+} from '@tds/core-colours'
+import { medium, mediumFont, color } from '@tds/shared-typography'
+import { FeedbackIcon } from '@tds/core-input'
+
+import safeRest from '../../shared/utils/safeRest'
```
Comment (theetrain) [inline packages/TextArea/TextArea.jsx]: Please add this to `devDependencies`

File: packages/TextArea/TextArea.jsx (hunk 2 above)
Comment (marcod1419) [reply packages/TextArea/TextArea.jsx]: It's possible to pass in tds links now.

File: packages/TextArea/TextArea.jsx (hunk 4 above)
Comment (marcod1419) [reply packages/TextArea/TextArea.jsx]: Confirmed that this is not a package in core.
